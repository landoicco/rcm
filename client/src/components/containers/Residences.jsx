import SingleResidence from "../model/SingleResidence";
import useDataSource from "../../hooks/useDataSource";
import { RESIDENCES_API_BASE_URL } from "../../commons/endpoints";

const Residences = () => {
  const residences = useDataSource(`${RESIDENCES_API_BASE_URL}/all`);
  return (
    <div className="container">
      <h3 className="center">Residences</h3>
      {residences.map((item) => (
        <SingleResidence key={item.id} item={item} />
      ))}
    </div>
  );
};

export default Residences;
