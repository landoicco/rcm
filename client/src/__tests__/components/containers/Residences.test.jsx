import { render, screen } from "@testing-library/react";
import Residences from "../../../components/containers/Residences";

describe("Residences", () => {
  test("The component render correctly", () => {
    const { container } = render(<Residences />);
    expect(container).toMatchSnapshot();
  });

  test("Should have only one heading element", async () => {
    render(<Residences />);

    // Get all elements with a 'heading' role
    const elements = await screen.findAllByRole("heading");
    expect(elements.length === 1).toBe(true);
  });

  test("The word 'Residences' should be rendered at least once", async () => {
    render(<Residences />);

    // Find all elements with the word 'Pets' on it
    const elements = await screen.findAllByText("Residences");
    expect(elements.length > 0).toBe(true);
  });
});
