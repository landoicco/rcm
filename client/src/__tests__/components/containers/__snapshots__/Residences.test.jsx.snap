// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Residences The component render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h3
      class="center"
    >
      Residences
    </h3>
  </div>
</div>
`;
